nixos/modules/services/networking/nsd.nix

```diff
@@ -930,20 +928,19 @@ in
         rm -Rf "${stateDir}/private/"
         rm -Rf "${stateDir}/tmp/"
 
-        mkdir -m 0700 -p "${stateDir}/private"
-        mkdir -m 0700 -p "${stateDir}/tmp"
-        mkdir -m 0700 -p "${stateDir}/var"
+        install -dm 0700 -o ${username} -g ${username} ${stateDir}/private
+        install -dm 0700 -o ${username} -g ${username} ${stateDir}/tmp
+        install -dm 0700 -o ${username} -g ${username} ${stateDir}/var
+
+        chown ${username}:${username} -R "${stateDir}/var"
```

Comment:
Two lines above, you create `${stateDir}/var` with the right owner and group. Is there a risk that this script is going to run against a state dir that already exists? If so, maybe it's worth adding a comment to explain this `chown`?

Also, this `chown` is happening before the `cat` below. Does that mean `${stateDir}/don't touch anything in here` will have the wrong owner/group?

Reply:
> Two lines above, you create ${stateDir}/var with the right owner and group. Is there a risk that this script is going to run against a state dir that already exists? If so, maybe it's worth adding a comment to explain this chown?

This chown here applied recursively to the files inside `$state/var` which might already exist. The install statement serve a different purpose, it's here to create the directory on the first run.

------------------------

> Also, this chown is happening before the cat below. Does that mean ${stateDir}/don't touch anything in here will have the wrong owner/group?

You're right, my bad. In the grand scheme of things, this comment is part of nixpkgs, a public git repo though. I'm not sure we can consider that as a secret.